contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -215,74 +215,70 @@ contract LoanTokenLogicStandard is LoanTokenSettingsLowerAdmin {
         uint256 collateralTokenSent,
         address collateralTokenAddress,
         address trader,
+		uint256 minReturn, // minimum position size in the collateral tokens
         bytes memory loanDataBytes // arbitrary order data
     )
         public
         payable
         nonReentrant //note: needs to be removed to allow flashloan use cases
         returns (
-            uint256,
-            uint256 // returns new principal and new collateral added to trade
+            uint256 newPrincipal,
+            uint256 newCollateral // returns new principal and new collateral added to trade
         )
     {
         _checkPause();
 
-        if (collateralTokenAddress == address(0)) {
-            collateralTokenAddress = wrbtcTokenAddress;
-        }
-
-        require(collateralTokenAddress != loanTokenAddress, "11");
-
-        //temporary: limit transaction size
-        if (transactionLimit[collateralTokenAddress] > 0)
-            require(
-                collateralTokenSent <= transactionLimit[collateralTokenAddress]
-            );
-        if (transactionLimit[loanTokenAddress] > 0)
-            require(loanTokenSent <= transactionLimit[loanTokenAddress]);
-
-        //computes the worth of the total deposit in loan tokens.
-        //(loanTokenSent + convert(collateralTokenSent))
-        //no actual swap happening here.
-        uint256 totalDeposit =
-            _totalDeposit(
-                collateralTokenAddress,
-                collateralTokenSent,
-                loanTokenSent
-            );
-        require(totalDeposit != 0, "12");
-
-        address[4] memory sentAddresses;
-        uint256[5] memory sentAmounts;
-
-        sentAddresses[0] = address(this); // lender
-        sentAddresses[1] = trader;
-        sentAddresses[2] = trader;
-        //sentAddresses[3] = address(0); // manager
-
-        //sentAmounts[0] = 0; // interestRate (found later)
-        sentAmounts[1] = totalDeposit; // total amount of deposit
-        //sentAmounts[2] = 0; // interestInitialAmount (interest is calculated based on fixed-term loan)
-        sentAmounts[3] = loanTokenSent;
-        sentAmounts[4] = collateralTokenSent;
-
-        _settleInterest();
-
-        (sentAmounts[1], sentAmounts[0]) = _getMarginBorrowAmountAndRate( // borrowAmount, interestRate
-            leverageAmount,
-            sentAmounts[1] // depositAmount
-        );
-
-        return
-            _borrowOrTrade(
-                loanId,
-                0, // withdrawAmount
-                leverageAmount,
-                collateralTokenAddress,
-                sentAddresses,
-                sentAmounts,
-                loanDataBytes
-            );
+		require(minReturn >= 0, "minReturn must larger than zero");
```

Comment:
it is `uint`, it is always greater or equal zero.  most probably, should be `require(minReturn > 0, ...)`